From ec2e2092bbc521ecf6d8d9d6d169acbb96625a9e Mon Sep 17 00:00:00 2001
From: "Gopal, Puunithaaraj" <puunithaaraj.gopal@intel.com>
Date: Wed, 8 May 2019 16:37:07 +0800
Subject: [PATCH 1395/1505] media: ici: Set mipi_store_mode to
 IPU_FW_ISYS_MIPI_STORE_MODE_DISCARD_LONG_HEADER

Set mipi_store_mode to IPU_FW_ISYS_MIPI_STORE_MODE_DISCARD_LONG_HEADER if using frontend capture

Change-Id: Ib3052234223ed3a8fcb7f31a90088e621fc9cd64
Tracked-On: PKT-2287
Signed-off-by: Puunithaaraj Gopal <puunithaaraj.gopal@intel.com>
---
 drivers/media/pci/intel/ici/ici-isys-stream.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/media/pci/intel/ici/ici-isys-stream.c b/drivers/media/pci/intel/ici/ici-isys-stream.c
index 4523ce9ed0b0..3187cae74d75 100644
--- a/drivers/media/pci/intel/ici/ici-isys-stream.c
+++ b/drivers/media/pci/intel/ici/ici-isys-stream.c
@@ -426,7 +426,6 @@ static int start_stream_firmware(struct ici_isys_stream *as)
 	stream_cfg.input_pins[0].input_res.height = source_fmt.ffmt.height;
 	stream_cfg.input_pins[0].dt =
 		ici_isys_format_code_to_mipi(source_fmt.ffmt.pixelformat);
-
 	/*
 	 * Only CSI2-BE has the capability to do crop,
 	 * so get the crop info from csi2-be.
@@ -452,7 +451,9 @@ static int start_stream_firmware(struct ici_isys_stream *as)
 			stream_cfg.crop[0].bottom_offset = ps.rect.top +
 				ps.rect.height;
 		}
-	}
+	} else
+		stream_cfg.input_pins[0].mipi_store_mode =
+			IPU_FW_ISYS_MIPI_STORE_MODE_DISCARD_LONG_HEADER;
 
 	as->prepare_firmware_stream_cfg(as, &stream_cfg);
 
-- 
2.17.1

